sky/tools/sky_snapshot/main.cc

```diff
@@ -63,5 +79,13 @@ int main(int argc, const char* argv[]) {
   CHECK(command_line.HasSwitch(switches::kSnapshot)) << "Need --snapshot";
   WriteSnapshot(command_line.GetSwitchValuePath(switches::kSnapshot));
 
+  if (command_line.HasSwitch(switches::kDepfile)) {
+    CHECK(command_line.HasSwitch(switches::kBuildOutput)) <<
+        "Need --build-output with --depfile";
+    WriteDependencies(command_line.GetSwitchValuePath(switches::kDepfile),
+                      command_line.GetSwitchValueASCII(switches::kBuildOutput),
```

Comment:
Ninja requires that the format of the dependency is 'target : list<dependencies>'. In our case the target is the .flx file not the intermediate snapshot.

I could also add a check for the case where 'build-output' is provided but 'depfile' isn't. But not sure if it'd be useful.